test/pageobjects/home.page.js

```diff
@@ -1,52 +1,95 @@
-    async getProductByName(productName) {
-        return await $(`//*[contains(text(), "${productName}")]`);
+  async clickProductByName(productName) {
+    const productElement = await this.getProductByName(productName)
+    await productElement.click()
+  }
+
+  async sortByPrice(direction) {
+    if (direction === 'Price (low to high)') {
+      await this.productSortContainer.selectByAttribute('value', 'lohi')
+    } else if (direction === 'Price (high to low)') {
+      await this.productSortContainer.selectByAttribute('value', 'hilo')
+    } else {
+      console.error('Invalid direction specified for sorting.')
     }
-    
+    return
+  }
 
-    async clickProductByName(productName) {
-        const productElement = await this.getProductByName(productName);
-        await productElement.click();
+  async sortByAlphabet(direction) {
+    if (direction === 'Name (A to Z)') {
+      await this.productSortContainer.selectByAttribute('value', 'az')
+    } else if (direction === 'Name (Z to A)') {
+      await this.productSortContainer.selectByAttribute('value', 'za')
+    } else {
+      console.error('Invalid direction specified for sorting.')
     }
+    return
+  }
 
+  async saveItemPrices() {
+    const priceElements = await $$('[data-test="inventory-item-price"]')
+    const prices = []
 
-    async sortByPrice(direction){
-        if(direction === "Price (low to high)"){
-            await this.productSortContainer.selectByAttribute('value', 'lohi');
-        } else if(direction === "Price (high to low)"){
-            await this.productSortContainer.selectByAttribute('value', 'hilo');
-        } else{
-            console.error('Invalid direction specified for sorting.');
-        }
-        return;
+    for (const priceElement of priceElements) {
+      const priceText = await priceElement.getText()
+      const priceValue = parseFloat(priceText.replace(/[^0-9.-]+/g, ''))
+      prices.push(priceValue)
     }
+    console.log('Sorted prices:', prices)
+    return prices
+  }
+
+  async saveItemNames() {
+    const nameElements = await $$('[data-test="inventory-item-name"]')
+    const names = []
 
-    async sortByAlphabet(direction){
-        if(direction === "Name (A to Z)"){
-            await this.productSortContainer.selectByAttribute('value', 'az');
-        } else if(direction === "Name (Z to A)"){
-            await this.productSortContainer.selectByAttribute('value', 'za');
-        } else{
-            console.error('Invalid direction specified for sorting.');
-        }
-        return;
+    for (const nameElement of nameElements) {
+      const nameText = await nameElement.getText()
+      names.push(nameText)
     }
+    console.log('Sorted names:', names)
+    return names
+  }
 
-    open () {
-        return super.open('inventory.html');
+  async isSorted(array, sortOption) {
+    if (
+      sortOption === 'Name (A to Z)' ||
+      sortOption === 'Price (low to high)'
```

Comment:
We use a sort of an enum here, so we should prefer a switch case if possible

Reply:
done